ethereum-explorer/src/map_contract_events.rs

```diff
@@ -0,0 +1,73 @@
+use crate::pb::eth::event::v1::Events;
+use crate::pb::eth::event::v1::Event;
+use substreams_ethereum::pb::eth::v2::Block;
+use substreams::Hex;
+use substreams_ethereum::pb::eth::v2::Log;
+use substreams_ethereum::pb::eth::v2::TransactionTrace;
+use substreams::errors::Error;
+use crate::util;
+
+#[substreams::handlers::map]
+fn map_contract_events(contract_address: String, blk: Block) -> Result<Events, Error> {
+   let error = verify_parameter(&contract_address);
+   if error.is_some() {
+    return Err(error.unwrap());
+   }
+
+    let mut events: Vec<Event> = Vec::new();
+    let contract_address_as_vec = match Hex::decode(&contract_address) {
+        Ok(address) => address,
+        Err(error) => return Err(Error::Unexpected(error.to_string())),
+    };
+
+    for transaction in &blk.transaction_traces {
+        if transaction.to == contract_address_as_vec {
+            let transaction_events = get_transaction_events(&transaction);
```

Comment:
Which will enable us to do:
```suggestion
            let transaction_events = get_transaction_events(transaction);
```